### PR TITLE
[Vertical compaction] improve the calculation method of aggregated rows in chunk aggregator

### PR DESCRIPTION
**Pull request by wyb**
The last row of the non-key column is in the column aggregator instead of in the aggregate chunk before finalizing.
In vertical compaction, there may be some only non-key columns in the aggregate chunk,
the _aggregate_chunk->num_rows() can not be used as the aggregate_rows.
So the aggregate_rows should be calculated in the aggregate function.

### CODE REVIEW

File: be/src/storage/vectorized/chunk_aggregator.h [hunk 1]
```diff
@@ -70,6 +70,7 @@ class ChunkAggregator {
 
     uint32_t _reserve_rows;
 
+    // total limit of aggregate rows
```
Comment (sduzh) [inline be/src/storage/vectorized/chunk_aggregator.h:73]: FYI: could you please give this variable another name, it's too similar to your newly added variable.

File: be/src/storage/vectorized/chunk_aggregator.h [hunk 2]
```diff
@@ -87,6 +88,10 @@ class ChunkAggregator {
     std::vector<uint32_t> _aggregate_loops;
 
     ChunkPtr _aggregate_chunk;
+    // the last row of non-key column is in aggregator (not in aggregate chunk) before finalize.
+    // in vertical compaction, there maybe only non-key column in aggregate chunk,
+    // so we cannot use _aggregate_chunk->num_rows() as aggregated rows.
+    uint32_t _aggregated_rows = 0;
```
Comment (sduzh) [inline be/src/storage/vectorized/chunk_aggregator.h:94]: Q: when will _aggregated_rows != _aggregate_chunk->num_rows()?

Comment (chaoyli) [inline be/src/storage/vectorized/chunk_aggregator.h:94]: what is the difference between _aggregate_rows and _aggregated_rows?

File: be/src/storage/vectorized/chunk_aggregator.h (hunk 1 above)
Comment (Seaven) [reply be/src/storage/vectorized/chunk_aggregator.h:73]: Agree with you

Comment (wyb) [reply be/src/storage/vectorized/chunk_aggregator.h]: ok

**Review by Seaven — DISMISSED** [review]

File: be/src/storage/vectorized/chunk_aggregator.h (hunk 2 above)
Comment (wyb) [reply be/src/storage/vectorized/chunk_aggregator.h]: Update to more clear names

Comment (wyb) [reply be/src/storage/vectorized/chunk_aggregator.h]: When there only non-key column in aggregate chunk, 
the value is in aggregator for sum(replace) and is not written to the column until a different key comes.

**Review by sduzh — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**